docs/docs/create-transformer-plugin.md

```diff
@@ -2,36 +2,273 @@
 title: Create a Transformer Plugin
 ---
 
-Transformer plugins are responsible for "transforming" data provided by
-[source plugins](/docs/create-source-plugin) into new nodes and/or node fields.
+There are two types of plugins that work within Gatsby's data system, "source"
+and "transformer" plugins.
+
+- **Source** plugins "source" data from remote or local locations into what
+  Gatsby calls [nodes](/docs/node-interface/).
+- **Transformer** plugins "transform" data provided by source plugins into new
+  nodes and/or node fields.
+
+This doc is concerned with transformer plugins.
 
 ## Transformer plugin example
 
-The [`gatsby-transformer-remark`](/packages/gatsby-transformer-remark/) plugin
-"transforms" markdown nodes (fetched for example from the filesystem source plugin) into html and provides additional node details such as for example the `timeToRead` the `tableOfContents` and the `excerpt`.
+The [`gatsby-transformer-remark`](/packages/gatsby-transformer-remark/) plugin "transforms" markdown nodes (fetched for example from the filesystem source plugin) into html and provides additional node details such as for example the `timeToRead` the `tableOfContents` and the `excerpt`.
 
 ## What do transformer plugins do?
 
-Each transformer plugin is responsible for "enhancing" the existing source nodes they
-are provided to provide additional information. Therefore, you'll often use both source plugins and transformer plugins in your Gatsby sites.
+Each transformer plugin is responsible for "enhancing" the existing source nodes they are provided to provide additional information. Therefore, you'll often use both source plugins and transformer plugins in your Gatsby sites.
```

Comment:
Enhancing isn't the right word. Transformers take data and we'll, "transform" it into a different type. Typically from something not JavaScript into JavaScript. Or something not html into html. Transformers can enhance data but that's an unusual case.

Reply:
Edited and rearranged a bit.